source/client.c

```diff
@@ -1022,17 +1036,6 @@ int aws_mqtt_client_connection_connect(
         aws_byte_buf_clean_up(&connection->client_id);
     }
 
-    /* Create the reconnect task for use later (probably) */
-    AWS_ASSERT(!connection->reconnect_task);
-    connection->reconnect_task = aws_mem_calloc(connection->allocator, 1, sizeof(struct aws_mqtt_reconnect_task));
-    if (!connection->reconnect_task) {
```

Comment:
down at the goto _failed labels of this function, there are 2 attempts in a row to free `connection->reconnect_task`. They need to be deleted